src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Order/TreeItemOrderPropertyProvider.cs

```diff
@@ -64,27 +75,66 @@ private void ComputeIndices()
             }
         }
 
+        /// <summary>
+        /// Tries to get a display order for a property context.
+        /// </summary>
+        /// <param name="propertyContext"></param>
+        /// <param name="displayOrder"></param>
+        /// <returns></returns>
+        private bool TryGetDisplayOrder(IProjectTreeCustomizablePropertyContext propertyContext, out int displayOrder)
+        {
+            // Find a display order using the item's name. This is not a relative path.
+            if (_displayOrderMap.TryGetValue(propertyContext.ItemName, out displayOrder))
+            {
+                return true;
+            }
+
+            // If we did not get a display order, we need to check its relative path.
+            // If the item has a valid Link property, use its value to find a display order.
+            var isLinked =
+                propertyContext.Metadata.TryGetValue(LinkProperty, out var linkPath) &&
+                !string.IsNullOrWhiteSpace(linkPath);
+            if (isLinked)
+            {
+                // Link path is always relative.
+                // Make sure we choose the property context with the same name as the last part of the link path.
+                if (string.Equals(propertyContext.ItemName, GetStringParts(linkPath).Last()))
+                {
+                    return _relativeDisplayOrderMap.TryGetValue(linkPath, out displayOrder);
+                }
+
+                // Sometimes, there can be two items with the same link path.
+                // Set the display order to zero on the one that doesn't match its name with its link path.
+                displayOrder = 0;
+                return true;
+            }
+
+            // If the item doesn't have a valid Link property, use its FullPath property value to find a display order.
+            // Full path needs to turn into a relative path for lookup.
+            return
+                propertyContext.Metadata.TryGetValue(FullPathProperty, out var fullPath) &&
+                _relativeDisplayOrderMap.TryGetValue(_project.MakeRelative(fullPath), out displayOrder);
```

Comment:
Above this was added as `Item.EvaluatedInclude`, now you've grabbed a normalized path relative to the project. What guarantee do they have to match in something like:

``` XML
<Compile Include="..\Ignored\..\Project\Program.cs" />
```

If you are storing paths in a dictionary for lookup - they have to be normalized.

Reply:
```
<Compile Include="..\Ignored\..\Project\Program.cs" />
```
Becomes a linked file with a path of "Program.cs". So, this is handled.

Reply:
What happens to these cases:

```
<Compile Include="Foo\..\Program.cs" />
<Compile Include="Foo\..\\Program.cs" />
<Compile Include="Foo/Program.cs" />
```

Reply:
From the second CPS PR, "Foo\..\Program.cs" and "Foo\..\\Program.cs" would return me "Program.cs".

"Foo/Program.cs" would return me "Foo/Program.cs"